vm/proxyapp/rpc/rpc.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2022 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package rpc
+
+import (
+	"time"
+
+	"github.com/google/syzkaller/pkg/report"
+)
+
+type ConstructPoolParams struct {
+	Debug bool
+	Param string
+}
+
+type ConstructPoolResult struct {
+	Count int // signal the created pool size
+}
+
+type CreateInstanceParams struct {
+	Workdir string
+	Index   int
+}
+
+type CreateInstanceResult struct {
+	ID string // allocated instance id
+}
+
+type MethodCopyParams struct {
+	ID      string
+	HostSrc string
+}
+
+type MethodCopyResult struct {
+	VMFileName string
+}
+
+type MethodForwardParams struct {
+	ID   string
+	Host string // currently is not used. Proxyapp use config param here.
+	Port int
+}
+
+type MethodForwardResult struct {
+	ManagerAddress string
+}
+
+type MethodRunStartParams struct {
+	ID      string
+	Timeout time.Duration
+	Command string
+}
+
+type MethodRunStartReply struct {
+	RunID string
+}
+
+type MethodRunStopParams struct {
+	ID    string
+	RunID string
+}
+
+type MethodRunStopReply struct {
+}
+
+type MethodRunReadProgressParams struct {
+	ID    string
+	RunID string
+}
+
+type MethodRunReadProgressReply struct {
+	StdoutStderr string
+	Error        string
+	Finished     bool
+	TimedOut     bool
+}
+
+type MethodCloseParams struct {
+	ID string
+}
+
+type MethodCloseReply struct {
+}
+
+type MethodDiagnoseParams struct {
+	ID     string
+	Reason report.Report
```

Comment:
If export this whole object here, then it become part of stable API that we can't change anymore. That's bad for us.
I would start with either not providing report here at all, or pass only Report.Title string. And then pass more info later if any plugins actually need it and we understand that this bit of info is stable and we can commit to providing it to external programs.

Reply:
Totally agree. Changed to Report.Title string.